### ISSUE TITLE
Feature request: start to debug c++ directly from command line.

### ISSUE THREAD

**Issue by kgfly** [microsoft/vscode-cpptools#3502]
Type: Debugger
**Input information below**

This a feature request. Currently, to debug the exe or attach a process, I've to setup launch.json, which is very tedious if you just want to do some quick very verification.

Request:  start to debug directly from command line, eg

code -e cpp -d ./a.out arg1 arg2, argN: to debug a.out with arg1 and arg2.... argN
code -e cpp -p 12345: attach process 12345.

"-e cpp" means "pass the following parameters to vscode-cpptools extension."

As a comparison:

In visual studio, you can devenv.exe /debugexe  a.exe

Eclipse has a plugin: https://github.com/eclipse-cdt/cdt/blob/master/debug/org.eclipse.cdt.debug.application/scripts/cdtdebug.sh, which allow you to start debug from command line.

Emacs (which is close to VScode since both VScode and emacs are editors while VS/Eclipse are IDE, you can pass some parameters https://stackoverflow.com/questions/5997309/start-emacs-gdb-from-command-line to start debugging directly.

**Comment by WardenGnaw** [comment]
This would be a feature for VS Code itself. There is no [commandline ](https://code.visualstudio.com/docs/editor/command-line)option to pass to the extension at the moment. Please submit an issue at https://github.com/Microsoft/vscode/issues

**Comment by 9527KGG** [comment]
https://github.com/Microsoft/vscode-cpptools/issues/3502 and https://github.com/Microsoft/vscode-python/issues/5396 are the related extension request. 
I think it needs work from vscode and extension.

**Comment by kgfly** [comment]
I think it need works from both vscode and extensions. I filed one in vscode. https://github.com/Microsoft/vscode/issues/72726

**Comment by WardenGnaw** [comment]
@9527KGG I agree that there may be work on our extension side. However, we are blocked until VS Code has a way to communicate to the extension that it needs to launch/attach a process from the commandline.

**Comment by kgfly** [comment]
Some one suggested a similar one but not for c++ though: https://marketplace.visualstudio.com/items?itemName=fabiospampinato.vscode-debug-launcher

**Comment by kgfly** [comment]
@WardenGnaw , https://marketplace.visualstudio.com/items?itemName=fabiospampinato.vscode-debug-launcher mention that they can debug script by running "vscode://fabiospampinato.vscode-debug-launcher/\<command\>?args=\<arg1\>,\<arg2\>,\<argN\>".
So can vscode-cpptools have similar feature? It does not require any work from vscode.

**Comment by WardenGnaw** [comment]
@9527KGG Thanks for letting us know. 

For this feature, this would require implementing VS Code's [UriHandler](https://github.com/Microsoft/vscode/blob/cb75ef9d0fd187ad1d2a75e9466b78092f105937/src/vs/vscode.d.ts#L6152) which should call [startDebugging](https://github.com/Microsoft/vscode/blob/cb75ef9d0fd187ad1d2a75e9466b78092f105937/src/vs/vscode.d.ts#L8867) with the given parameters.

**Comment by eternalphane** [comment]
Hi @9527KGG, why you close this issue?

**Comment by kgfly** [comment]
> Hi @9527KGG, why you close this issue?

because I gave up :)

**Comment by eternalphane** [comment]
> > Hi @9527KGG, why you close this issue?
> 
> because I gave up :)

then please reopen this issue to mark it as unresolved (and hopefully someone may create a pr for it ˊ_>ˋ )

**Comment by weiguozhi** [comment]
This feature is also useful to me. Any progress in the past two years?

I saw many similar feature requests were marked as duplication of https://github.com/microsoft/vscode/issues/10979, and 10979 was closed recently. But in 10979 I can't find anything useful to a task described in this issue.
